chainstate/src/detail/mod.rs

```diff
@@ -268,13 +260,35 @@ impl Chainstate {
         self.attempt_to_process_block(block, block_source, 0)
     }
 
+    /// Initialize chainstate with genesis block
+    pub fn process_genesis(&mut self) -> Result<(), BlockError> {
+        use chainstate_storage::{BlockchainStorageWrite, TransactionRw};
+
+        // Gather information about genesis.
+        let genesis = self.chain_config.genesis_block();
+        let genesis_id = self.chain_config.genesis_block_id();
+        let utxo_count = genesis.utxos().len() as u32;
+        let genesis_index =
+            common::chain::TxMainChainIndex::new(genesis_id.clone().into(), utxo_count)
+                .expect("Genesis not constructed correctly");
+
+        // Initialize storage with given info
+        let mut tx = self.chainstate_storage.transaction_rw();
```

Comment:
I know it's nothing... but please call it `db_tx`... we wanna try to always distinguish between a value transaction and a database transaction that way.